src/lib/widget/_utilities/api.utilities.ts

```diff
@@ -1,254 +1,213 @@
-          returnError &&
-            returnError({
-              journey: response,
-            });
-        }
+  function resetAndRestartStores() {
+    // Reset stores
+    journeyStore && journeyStore.reset();
+    oauthStore && oauthStore.reset();
+    userStore && userStore.reset();
+
+    // Fetch fresh journey step
+    journey && journey().start();
+  }
+
+  interface widgetConfigOptions {
+    config: z.infer<typeof partialConfigSchema> | undefined;
+    content: z.infer<typeof partialStringsSchema> | undefined;
+    journeys: z.infer<typeof journeyConfigSchema> | undefined;
+    links: z.infer<typeof partialLinksSchema> | undefined;
+    style: z.infer<typeof partialStyleSchema> | undefined;
+  }
+
+  const configuration = () => {
+    return {
+      set(options: widgetConfigOptions): void {
+        configure({
+          // Set some basics by default
+          ...{
+            // TODO: Could this be a default OAuth client provided by Platform UI OOTB?
+            clientId: 'WebLoginWidgetClient',
+            // TODO: If a realmPath is not provided, should we call the realm endpoint and detect a likely default?
+            // https://backstage.forgerock.com/docs/am/7/setup-guide/sec-rest-realm-rest.html#rest-api-list-realm
+            realmPath: 'alpha',
+            // TODO: Once we move to SSR, this default should be more intelligent
+            redirectUri:
+              typeof window === 'object' ? window.location.href : 'https://localhost:3000/callback',
+            scope: 'openid email',
+          },
+          // Let user provided config override defaults
+          ...options.config,
+          // Force 'legacy' to remove confusion
+          ...{ support: 'legacy' },
+        });
+
         /**
-         * Clean up unneeded subscription, but only when it's successful
-         * Leaving the subscription allows for the journey to be
-         * restarted internally.
+         * Initialize the stores and ensure both variables point to the same reference.
+         * Variables with _ are the reactive version of the original variable from above.
          */
-        if (response.successful) {
-          journeyStoreUnsub();
-        }
-      });
+        journeyStore = initializeJourney(options.config);
+        oauthStore = initializeOauth(options.config);
+        userStore = initializeUser(options.config);
+
+        initializeContent(options.content);
+        initializeJourneys(options.journeys);
+        initializeLinks(options.links);
+        initializeStyle(options.style);
+      },
+    };
+  };
+  const journey = (options?: JourneyOptions) => {
+    const requestsOauth = options?.oauth || true;
+    const requestsUser = options?.user || true;
+    const { subscribe } = derived(
+      [journeyStore, oauthStore, userStore],
+      ([$journeyStore, $oauthStore, $userStore], set) => {
+        set({
+          journey: $journeyStore,
+          oauth: $oauthStore,
+          user: $userStore,
+        });
```

Comment:
Here, we're using a `derived` store to create a single store out of the three stores, `journey`, `oauth`, and `user`. This single store will emit an event when any of its child stores emit an event. We have complete and fine-grain control over what we emit to the developer, from an event standpoint and a data standpoint. It can be as much or as little as we want.